pandas/src/algos_groupby_helper.pxi.in

```diff
@@ -669,6 +669,73 @@ def group_cumsum(numeric[:, :] out,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def group_cummin(numeric[:, :] out,
+                 numeric[:, :] values,
+                 int64_t[:] labels,
+                 numeric[:, :] accum):
+    """
+    Only transforms on axis=0
+    """
+    cdef:
+        Py_ssize_t i, j, N, K, size
+        numeric val, min_val
+        int64_t lab
+
+    N, K = (<object> values).shape
+    accum = np.zeros_like(accum)
+
+    with nogil:
+        for i in range(N):
+            lab = labels[i]
+
+            if lab < 0:
+                continue
+            for j in range(K):
+                val = values[i, j]
+                if val == val:
+                    if j == 0:
+                        min_val = val
+                    elif val < min_val:
```

Comment:
Ah thanks! Was doing some local tests and noticed the results didn't exactly match the alternative `groupby.apply(np.minimum.accumulate)`. Will check this out tonight.